business_requirement/models/business.py

```diff
@@ -0,0 +1,275 @@
+    )
+    state = fields.Selection(
+        selection="_get_states",
+        string='State',
+        default='draft',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        track_visibility='onchange'
+    )
+    business_requirement_ids = fields.One2many(
+        comodel_name='business.requirement',
+        inverse_name='parent_id',
+        string='Sub Business Requirement',
+        copy=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    parent_id = fields.Many2one(
+        comodel_name='business.requirement',
+        string='Parent',
+        ondelete='set null',
+        domain="[('id', '!=', id)]",
+        readonly=True,
+        states={
+            'draft': [('readonly', False)],
+            'confirmed': [('readonly', False)]}
+    )
+    level = fields.Integer(
+        compute='_get_level',
+        string='Level',
+        store=True
+    )
+    change_request = fields.Boolean(
+        string='Change Request?',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    project_id = fields.Many2one(
+        comodel_name='project.project',
+        string='Master Project',
+        ondelete='set null',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Customer',
+        store=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    sub_br_count = fields.Integer(
+        string='Count',
+        compute='_compute_sub_br_count'
+    )
+    priority = fields.Selection(
+        [('0', 'Low'), ('1', 'Normal'), ('2', 'High')],
+        'Priority',
+        required=True,
+        default='1'
+    )
+    requested_id = fields.Many2one(
+        'res.users',
+        string='Requested by',
+        required=True,
+        readonly=True,
+        default=lambda self: self.env.user,
+        states={
+            'draft': [('readonly', False)],
+            'confirmed': [('readonly', False)]}
+    )
+    confirmation_date = fields.Datetime(
+        string='Confirmation Date',
+        readonly=True
+    )
+    confirmed_id = fields.Many2one(
+        'res.users', string='Confirmed by',
+        readonly=True
+    )
+    reviewed_date = fields.Datetime(
+        string='Reviewed Date',
+        readonly=True
+    )
+    reviewed_id = fields.Many2one(
+        'res.users', string='Reviewed by',
+        readonly=True
+    )
+    approval_date = fields.Datetime(
+        string='Approval Date',
+        readonly=True
+    )
+    approved_id = fields.Many2one(
+        'res.users',
+        string='Approved by',
+        readonly=True
+    )
+    company_id = fields.Many2one(
+        'res.company', string='Company',
+        required=True, readonly=True, states={'draft': [('readonly', False)]},
+        default=_get_default_company,
```

Comment:
What about use the following way https://github.com/odoo/odoo/blob/b2804b1d2ae5fd7ec5c0eca79a8db77476badc84/addons/account/models/account.py#L192?

`company_id = fields.Many2one('res.company', string='Company', required=True, index=True, default=lambda self: self.env.user.company_id,
        help="Company related to this journal")`